### PR TITLE
Add ConstAllocator and static functions create(), ZEROBUF_TYPE_NAME(), ZEROBUF_TYPE_IDENTIFIER()

### PR DESCRIPTION
**Pull request by tribal-tec**
This is part of event-based communication spec from HBPVIS/ZeroEQ#152.

### CODE REVIEW

File: zerobuf/ConstAllocator.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    const uint8_t* getData() const final { return _data; }
+    size_t getSize() const final { return _size; }
+
+private:
+    ConstAllocator( const ConstAllocator& ) = delete;
+    ConstAllocator& operator = ( const ConstAllocator& ) = delete;
+
+    uint8_t* getData() final
+        { return const_cast< uint8_t* >( _data ); } // internal
```
Comment (hernando) [inline zerobuf/ConstAllocator.h]: Be aware that this is not sufficient to protect the internal buffer. If the allocator is used through it's base class this function is not private. I'm not sure making it throw prevents non const access, without reading all the client code is difficult to say if the compiler is going to pick always the const version.

Comment (tribal-tec) [inline zerobuf/ConstAllocator.h]: I had the throw before for the same reasons, but it does not work unfortunately. With the test schema in the unit tests, it passes by code which calls getData() and the compiler prefers the non-const over the const version, hence it passes by here. A delete for this non-const getData() is what one wants here...

File: tests/constAllocator.cpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    BOOST_CHECK_THROW( alloc->updateAllocation( 1, 2, 3 ), std::runtime_error );
+    BOOST_CHECK_THROW( alloc->compact( 42 ), std::runtime_error );
+}
+
+BOOST_AUTO_TEST_CASE(create)
+{
+    test::TestSchema object;
+
+    const auto& data = object.toBinary();
+    test::ConstTestSchemaPtr newObject = test::TestSchema::create( data.ptr.get(), data.size );
```
Comment (hernando) [inline tests/constAllocator.cpp]: The code would look better if the only valid syntax was:

    const test::TestSchema newConstObject = test::TestSchema::fromBinary( data.ptr.get(), data.size );

But doesn't look like something possible in C++.
Still I'd rename it to fromBinary.

File: bin/zerobufCxx.py [hunk 3]
```diff
@@ -958,8 +962,12 @@ def introspection_functions(self):
         functions = []
         functions.append(Function("std::string", "getTypeName() const final",
                                   "return \"{0}\";".format(zerobufName), split=False))
```
Comment (hernando) [inline bin/zerobufCxx.py]: You can make these lines more readable if you combine " and '

    'return "{0}";'.format(zerobufName)

I also find the version below even more readable, but that's a matter of taste:

    'return "%s";' % zerobufName

**Comment by hernando** [comment]
Besides the cosmetic changes about strings +1

**Pull request merged**